am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -58,35 +65,50 @@ void whenCreatingResourceAccessMultipleEntriesAppearInDatabase() {
             JsonPointer.valueOf(""), ImmutableSet.of(CREATE, READ, UPDATE),
             JsonPointer.valueOf("/name"), ImmutableSet.of(CREATE, READ, UPDATE));
 
-        service.grantExplicitResourceAccess(createGrant(resourceId, accessorId, resourceDefinition, multipleAttributePermissions));
+        service.grantExplicitResourceAccess(createGrant(
+            resourceId, accessorId, relationship, resourceDefinition, multipleAttributePermissions));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(2);
     }
 
     @Test
     void whenCreatingDuplicateResourceAccessEntryIsOverwritten() {
-        service.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, accessorId, resourceDefinition, ImmutableSet.of(READ)));
-        service.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, accessorId, resourceDefinition, ImmutableSet.of(READ)));
+        service.grantExplicitResourceAccess(createGrantForWholeDocument(
+            resourceId, accessorId, relationship, resourceDefinition, ImmutableSet.of(READ)));
+        service.grantExplicitResourceAccess(createGrantForWholeDocument(
+            resourceId, accessorId, relationship, resourceDefinition, ImmutableSet.of(READ)));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
     }
 
     @Test
     void whenCreatingResourceForMultipleUsersShouldAppearInDatabase() {
-        service.grantExplicitResourceAccess(
-            createGrantForWholeDocument(resourceId, ImmutableSet.of("User1", "User2"), resourceDefinition, ImmutableSet.of(READ)));
+        service.grantExplicitResourceAccess(createPermissionsForWholeDocumentForMultipleUsers(
```

Comment:
Why you pass permissions to this private method if tests does not care about permissions? Resource ID in turn that is important is hidden :(

Reply:
👍